clients/admin-ui/src/features/messaging-templates/AddMessagingTemplateModal.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Button,
+  ButtonGroup,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Select,
+  Text,
+} from "fidesui";
+import { useState } from "react";
+
+import { CustomizableMessagingTemplatesEnum } from "~/features/messaging-templates/CustomizableMessagingTemplatesEnum";
+import CustomizableMessagingTemplatesLabelEnum from "~/features/messaging-templates/CustomizableMessagingTemplatesLabelEnum";
+
+interface AddMessagingTemplateModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onAccept: (templateId: string) => void;
+}
+
+const AddMessagingTemplateModal: React.FC<AddMessagingTemplateModalProps> = ({
+  isOpen,
+  onClose,
+  onAccept,
+}) => {
+  const [selectedTemplateId, setSelectedTemplateId] = useState<
+    string | undefined
+  >(undefined);
+
+  const messagingActionTypeIds = Object.keys(
+    CustomizableMessagingTemplatesLabelEnum
+  ) as CustomizableMessagingTemplatesEnum[];
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size="2xl" isCentered>
+      <ModalOverlay />
+      <ModalContent data-testid="add-messaging-template-modal">
+        <ModalHeader borderBottomWidth={1} fontWeight="bold">
+          Select message template
+        </ModalHeader>
+        <ModalBody>
+          <Text
+            color="gray.700"
+            fontWeight="medium"
+            fontSize="sm"
+            marginBottom={3}
+            marginTop={1}
+          >
+            Add a new email message by selecting a template below and clicking
+            accept.
+          </Text>
+          <Text
+            color="gray.700"
+            fontSize="sm"
+            fontWeight="medium"
+            marginBottom={2}
+          >
+            Choose template:
+          </Text>
+          <Select
```

Comment:
Can we use the select styles that we use elsewhere in the app for this?  The needed Chakra styles are in a const `SELECT_STYLES` in inputs.tsx, and doing so would also let you use the `enumToOptions` helper method we have for cases like this.

Reply:
I've updated the modal to use the react-select selector. I didn't use enumToOptions because I needed the value and the label to be different.